src/go/rpk/pkg/cli/container/start.go

```diff
@@ -66,60 +68,51 @@ func newStartCommand() *cobra.Command {
 			// (POSIX standard)
 			UnknownFlags: true,
 		},
-		RunE: func(_ *cobra.Command, _ []string) error {
+		Run: func(cmd *cobra.Command, _ []string) {
 			if nodes < 1 {
-				return errors.New(
-					"--nodes should be 1 or greater",
-				)
-			}
-			c, err := common.NewDockerClient()
-			if err != nil {
-				return err
+				out.Die("--nodes should be 1 or greater")
 			}
+			c, err := common.NewDockerClient(cmd.Context())
+			out.MaybeDie(err, "unable to create docker client: %v", err)
 			defer c.Close()
 
 			configKvs := collectFlags(os.Args, "--set")
+			isRestarted, err := startCluster(c, nodes, checkBrokers, retries, image, pull, configKvs)
+			out.MaybeDieErr(common.WrapIfConnErr(err))
 
-			return common.WrapIfConnErr(startCluster(
-				c,
-				nodes,
-				checkBrokers,
-				retries,
-				image,
-				pull,
-				configKvs,
-			))
+			if noProfile || isRestarted {
+				return
+			}
+
+			dockerNodes, err := renderClusterInfo(c)
+			out.MaybeDie(err, "unable to render cluster info: %v; you may run 'rpk container status' to retrieve the cluster info", err)
+
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			y, err := cfg.ActualRpkYamlOrEmpty()
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			err = common.CreateProfile(fs, c, y)
+			if err == nil {
+				fmt.Printf("\nCreated %q profile.\n", common.ContainerProfileName)
+				renderClusterInteract(dockerNodes, true)
+				return
+			}
+			if errors.Is(err, common.ErrContainerProfileExists) {
+				fmt.Printf("Unable to create a profile for the rpk container: %v; you may delete it and create a new one running 'rpk profile create --from-rpk-container'\n", err)
+				return
+			} else {
+				out.Die("unable to create a profile for the rpk container: %v", err)
+			}
 		},
 	}
 
-	command.Flags().UintVarP(
-		&nodes,
-		"nodes",
-		"n",
-		1,
-		"The number of nodes to start",
-	)
-
-	command.Flags().UintVar(
-		&retries,
-		"retries",
-		10,
-		"The amount of times to check for the cluster before"+
-			" considering it unstable and exiting",
-	)
-	imageFlag := "image"
-	command.Flags().StringVar(
-		&image,
-		imageFlag,
-		common.DefaultImage(),
-		"An arbitrary container image to use",
-	)
-	command.Flags().BoolVar(
-		&pull,
-		"pull",
-		false,
-		"Force pull the container image used",
-	)
+	command.Flags().UintVarP(&nodes, "nodes", "n", 1, "The number of nodes to start")
+	command.Flags().UintVar(&retries, "retries", 10, "The amount of times to check for the cluster before considering it unstable and exiting")
+	command.Flags().StringVar(&image, "image", common.DefaultImage(), "An arbitrary container image to use")
+	command.Flags().BoolVar(&pull, "pull", false, "Force pull the container image used")
+	command.Flags().BoolVar(&noProfile, "no-profile", false, "If true, rpk will not create an rpk profile after creating a cluster")
```

Comment:
Do we want to use no-profile as a global flag to disable using profiles? Or should we use something else?

Reply:
:thinking: Maybe use here `--no-profile-creation`.

A little bit longer but I don't expect this flag to have a lot of usage tbh, and this way we leave the room open for a global `--no-profile`

wdyt?